018-inertiajs.md

```diff
@@ -0,0 +1,97 @@
+
+In his response he also mentions the following:
+
+> Finally, at the end of the day Inertia is actually a pretty thin layer between your app's back-end and front-end. If you ever decided that Inertia wasn't working for you, removing it would be a relatively trivial task.
+
+I thought this was insightful and knowing that if one day we changed our minds, removing Inertia JS from our code would not necessarily be a heavy lift (most of the work would be in finding/building a different solution to connect our data to the React frontend).
+
+## Alternatives
+
+There are a couple of alternatives we also considered, but with either one it would potentially mean either less flexibility, or lots of overhead and more maintenance to enable our existing backend code to communicate with the frontend.
+
+### Laravel Nova
+
+[Laravel Nova](https://nova.laravel.com/) is a one-time purchase that essentially provides a full administrative panel for a Laravel application.
+
+It has a whole slew of features that are very compelling. The main benefit Nova provides is that the team would not have to do much work building out an administrative panel.
+
+However, I am cautious that how our Eloquent models have evolved and the different relationships we have established between resources, combined with the merging of services may not completely follow the Laravel approach and we could run into numerous headaches.
+
+I also imagine we would have to wait until we complete the unified database and switch off MongoDB completely before we could move to use Nova, to avoid issues since the MongoDB Eloquent driver is a community contribution and not officially supported by Laravel.
+
+Nova does offer CLI generators to create custom tools within the framework that could support our business needs. The downside is that these tools are generated as Vue components, which would require our developers to learn [Vue JS](https://vuejs.org/) and context switch between Vue JS for administrative tools and React JS for member facing tools. While the need for custom tools may be low, it would imply additional maintenance overhead in a client-side framework the team is not familiar with.
+
+### React Admin
+
+[React Admin](https://marmelab.com/react-admin) is an open-source framework for building admin applications in React with data provided by REST/GraphQL APIs.
+
+The framework looks like it has a number of useful components to help build out an administrative interface. The main benefit React Admin provides is that the team would not have to do much work building out components for an administrative interface.
+
+However, the framework is dependent on having established API endpoints for all the resources it will render. This would require the team to invest time and effort into building out endpoints for all backend resources that would be administered via a frontend UI and maintaining all the endpoints over time.
```

Comment:
Is the reverse at all a concern -- do we have any API-based resources from Rogue that we'd need to build new route/controllers for to enable Intertia? (And/or perhaps that's an upside if philosophically we'd rather have classic web routes for non-external-based needs?)

Reply:
I didn't do a complete review of the Rogue, now Northstar code regarding this, but given a lot of the resources were available to the front-end of the Rogue Admin I think any API-only resources were very few!

Reply:
As far as I know, our ~Rogue~ Northstar admin UIs rely on the `v3/` API endpoints (via GraphQL) for most interfaces so we would have to create new Inertia controllers for all of those interfaces.

Reply:
Ah right, good call! 

But that's for resources now native to Northstar for the most part so I think it'd be pretty straightforward and eliminate a roundtrip to a different service for a resource that would be used for the UI. I think in the long run it would reduce the overhead of our GraphQL service if GraphQL only cared about and needs to handle shared service resources and not for resources used mostly for Admin purposes (oversimplifying it, but I hope that makes sense!)